packages/vite-plugin/src/proxy-middleware.ts

```diff
@@ -0,0 +1,244 @@
+/**
+ * This modules provides the proxy located at
+ * /motion-canvas-proxy/...
+ *
+ * It is needed when acessing remote resources.
+ * Trying to access remote resources works while
+ * in preview, but will fail when you try to
+ * output the image (= "read" the canvas)
+ *
+ * See https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_enabled_image
+ * for reasons
+ *
+ * Using the proxy circumvents CORS-issues because
+ * this way all remote resources are served from the
+ * same host as the main app.
+ */
+
+import {Connect} from 'vite';
+import axios from 'axios';
+import {ServerResponse, IncomingMessage} from 'http';
+
+/**
+ * Configuration used by the Proxy plugin
+ */
+export interface MotionCanvasCorsProxyPluginOptions {
+  /**
+   * Set which types of resources are allowed by default.
+   *
+   * @remarks
+   * Catchall on the right side is supported.
+   * Pass an empty Array to allow all types of resources, although this is not recommended.
```

Comment:
Prettier won't do this for you but comments should wrap at 80 characters too.
```suggestion
   * Pass an empty Array to allow all types of resources, although this is not 
   * recommended.
```